File: cp3407-project-main/webpage/html_design/submit_registration.php
<?php

$servername = "localhost";
$username = "your_username";
$password = "your_password";
$dbname = "user_registration";

$conn = new mysqli($servername, $username, $password, $dbname);


if ($conn->connect_error) {
    die("连接失败: " . $conn->connect_error);
}


$sql = "INSERT INTO users (fullname, email, phone, username, password, address, city, zip) VALUES (?, ?, ?, ?, ?, ?, ?, ?)";


$stmt = $conn->prepare($sql);
$stmt->bind_param("ssssssss", $fullname, $email, $phone, $username, $password_hash, $address, $city, $zip);


$fullname = $_POST['fullname'];
$email = $_POST['email'];
$phone = $_POST['phone'];
$username = $_POST['username'];
$password = $_POST['password'];
$password_hash = password_hash($password, PASSWORD_DEFAULT); // 对密码进行哈希加密以提高安全性
$address = $_POST['address'];
$city = $_POST['city'];
$zip = $_POST['zip'];


if ($stmt->execute()) {
    echo "注册成功！";
} else {
    echo "错误: " . $sql . "<br>" . $conn->error;
}


$stmt->close();
$conn->close();
?>

File: cp3407-project-main/webpage/html_design/submit_registration.py
import cgi
import mysql.connector

# 连接到数据库
db_connection = mysql.connector.connect(
    host="localhost",
    user="root",
    password="Songjiahao4396",
    database="user_information"
)
print("Database connection successful.")


form = cgi.FieldStorage()
fullname = "John Doe"
email = "john.doe@example.com"
phone = "1234567890"
username = "johndoe"
password = "secure_password"
address = "123 Main St"
city = "Anytown"
zip_code = "12345"
# fullname = form.getvalue('fullname')
# email = form.getvalue('email')
# phone = form.getvalue('phone')
# username = form.getvalue('username')
# password = form.getvalue('password')
# address = form.getvalue('address')
# city = form.getvalue('city')
# zip_code = form.getvalue('zip')
print("get form")


cursor = db_connection.cursor()
insert_query = """
    INSERT INTO users (fullname, email, password, username, address, city, zip)
    VALUES (%s, %s, %s, %s, %s, %s, %s)
"""
insert_values = (fullname, email, password, username, address, city, zip_code)
cursor.execute(insert_query, insert_values)


db_connection.commit()


db_connection.close()


print("Content-type:text/html\r\n\r\n")
print("<html>")
print("<head>")
print("<title>Registration Successful</title>")
print("</head>")
print("<body>")
print("<h2>Registration Successful</h2>")
print("<p>Thank you for registering.</p>")
print("</body>")
print("</html>")